pkg/tarsum/versioning.go

```diff
@@ -115,25 +115,26 @@ func v0TarHeaderSelect(h *tar.Header) (orderedHeaders [][2]string) {
 
 func v1TarHeaderSelect(h *tar.Header) (orderedHeaders [][2]string) {
 	// Get extended attributes.
-	xAttrKeys := make([]string, len(h.Xattrs))
-	for k := range h.Xattrs {
-		xAttrKeys = append(xAttrKeys, k)
+	const paxSchilyXattr = "SCHILY.xattr."
+	var xattrs [][2]string
+	for k, v := range h.PAXRecords {
+		if xattr, ok := strings.CutPrefix(k, paxSchilyXattr); ok {
+			xattrs = append(xattrs, [2]string{xattr, v})
```

Comment:
Ah, smart to construct the key/value pair here instead of further down (I was looking here as to "what changed?" but now see this is to prevent the loop at line 134 👍

Reply:
Interesting; looks like BuildKit forked this code, and now I'm wondering if the implementation should be shared (the tarsum specification looks to be in this package); https://github.com/moby/moby/blob/452ca90fe5f010f96fd50bb0fa9622e6e3d1e50d/vendor/github.com/moby/buildkit/cache/contenthash/tarsum.go#L38-L76